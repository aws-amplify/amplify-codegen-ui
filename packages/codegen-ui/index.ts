/*
  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.

  Licensed under the Apache License, Version 2.0 (the "License").
  You may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */
export * from './lib/template-renderer';
export * from './lib/frontend-manager-template-renderer';

export * from './lib/component-renderer-base';
export * from './lib/component-with-children-renderer-base';
export * from './lib/render-component-response';
export * from './lib/framework-output-manager';
export * from './lib/template-renderer-factory';

export * from './lib/renderer-helper';
export * from './lib/validation-helper';
export * from './lib/framework-output-config';
export * from './lib/framework-render-config';
export * from './lib/frontend-manager-node';

export * from './lib/types';
export * from './lib/errors';
