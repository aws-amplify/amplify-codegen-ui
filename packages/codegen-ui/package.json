{
  "name": "@aws-amplify/codegen-ui",
  "version": "2.20.2",
  "description": "generic component code generation interface definitions",
  "author": "Amazon Web Services",
  "homepage": "https://docs.amplify.aws/",
  "repository": "https://github.com/aws-amplify/amplify-codegen-ui.git",
  "license": "Apache-2.0",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist/**"
  ],
  "scripts": {
    "test": "jest",
    "test:ci": "jest -i --ci",
    "test:update": "jest --updateSnapshot",
    "build": "tsc -p tsconfig.build.json",
    "build:watch": "npm run build -- --watch"
  },
  "dependencies": {
    "change-case": "^4.1.2",
    "yup": "^0.32.11"
  },
  "devDependencies": {
    "@aws-amplify/appsync-modelgen-plugin": "^2.3.3",
    "@aws-amplify/datastore": "^4.0.15"
  },
  "jest": {
    "verbose": false,
    "preset": "ts-jest",
    "transform": {
      "^.+\\.tsx?$": "ts-jest"
    },
    "collectCoverage": true,
    "collectCoverageFrom": [
      "lib/**/*.ts"
    ],
    "coveragePathIgnorePatterns": [
      "<rootDir>/lib/__tests__/__utils__/"
    ],
    "testRegex": "(lib/__tests__/.*.test.ts)$",
    "testPathIgnorePatterns": [
      "<rootDir>/lib/__tests__/__utils__/"
    ],
    "globals": {
      "ts-jest": {
        "diagnostics": false
      }
    },
    "moduleFileExtensions": [
      "ts",
      "js"
    ]
  },
  "overrides": {
    "hermes-engine": ">=0.10.0",
    "shell-quote": ">=1.7.3"
  }
}
