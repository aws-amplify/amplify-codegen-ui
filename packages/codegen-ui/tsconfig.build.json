{
  "extends": "./tsconfig.json",
  "exclude": [
    "lib/__tests__/*"
  ]
}
