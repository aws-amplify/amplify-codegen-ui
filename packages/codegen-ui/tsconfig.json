{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,
    "declarationMap": true,
    "composite": true,
    "outDir": "./dist",
    "sourceMap": true,
    "skipLibCheck": true
  },
  "include": ["./lib", "index.ts"],
  "exclude": ["dist"]
}
