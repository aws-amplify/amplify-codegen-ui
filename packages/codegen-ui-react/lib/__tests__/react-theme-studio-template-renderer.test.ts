/*
  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.

  Licensed under the Apache License, Version 2.0 (the "License").
  You may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */
import { StudioTemplateRendererFactory, StudioTheme } from '@aws-amplify/codegen-ui';
import { ScriptTarget, ScriptKind, ReactRenderConfig } from '..';
import {
  ReactThemeStudioTemplateRenderer,
  ReactThemeStudioTemplateRendererOptions,
} from '../react-theme-studio-template-renderer';
import { loadSchemaFromJSONFile } from './__utils__';

function generateWithThemeRenderer(
  jsonFile: string,
  renderConfig: ReactRenderConfig = {},
  options?: ReactThemeStudioTemplateRendererOptions,
) {
  const rendererFactory = new StudioTemplateRendererFactory(
    (theme: StudioTheme) => new ReactThemeStudioTemplateRenderer(theme, renderConfig, options),
  );
  const { componentText, declaration } = rendererFactory
    .buildRenderer(loadSchemaFromJSONFile(jsonFile))
    .renderComponent();
  return { componentText, declaration };
}

function generateThemeObject(jsonFile: string): any {
  const rendererFactory = new StudioTemplateRendererFactory(
    (theme: StudioTheme) => new ReactThemeStudioTemplateRenderer(theme, {}),
  );
  const themeJson = rendererFactory.buildRenderer(loadSchemaFromJSONFile(jsonFile)).renderThemeJson();
  /* eslint-disable @typescript-eslint/no-implied-eval */
  const themeObject = new Function(`return ${themeJson}`);
  /* eslint-enable @typescript-eslint/no-implied-eval */
  return themeObject();
}

describe('react theme renderer tests', () => {
  describe('theme', () => {
    it('should render the theme', () => {
      expect(generateWithThemeRenderer('theme').componentText).toMatchSnapshot();
    });

    it('should render the theme with TSX', () => {
      const { componentText, declaration } = generateWithThemeRenderer('theme', { script: ScriptKind.TSX });
      expect(componentText).toMatchSnapshot();
      expect(declaration).toBeUndefined();
    });

    it('should render the theme with ES5', () => {
      const { componentText, declaration } = generateWithThemeRenderer('theme', {
        target: ScriptTarget.ES5,
        script: ScriptKind.JS,
        renderTypeDeclarations: true,
      });
      expect(componentText).toMatchSnapshot();
      expect(declaration).toBeDefined();
      expect(declaration).toMatchSnapshot();
    });

    it('should render the default theme', () => {
      expect(generateWithThemeRenderer('theme', {}, { renderDefaultTheme: true }).componentText).toMatchSnapshot();
    });
  });

  describe('renderThemeJson', () => {
    it('should render theme json successfully', () => {
      const rendererFactory = new StudioTemplateRendererFactory(
        (theme: StudioTheme) => new ReactThemeStudioTemplateRenderer(theme, {}),
      );
      expect(rendererFactory.buildRenderer(loadSchemaFromJSONFile('theme')).renderThemeJson()).toMatchSnapshot();
    });

    it('should render theme json with breakpoints successfully', () => {
      const themeObject = generateThemeObject('themeWithBreakpoints');
      expect(themeObject).toBeDefined();
      expect(typeof themeObject.breakpoints.values.base).toBe('number');
      expect(themeObject.breakpoints.values.base).toEqual(0);
      expect(typeof themeObject.breakpoints.defaultBreakpoint).toBe('string');
      expect(themeObject.breakpoints.defaultBreakpoint).toBe('base');
    });
  });
});
