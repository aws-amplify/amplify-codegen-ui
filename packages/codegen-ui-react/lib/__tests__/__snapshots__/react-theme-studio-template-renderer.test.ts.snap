// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react theme renderer tests renderThemeJson should render theme json successfully 1`] = `
"{
  name: \\"MyTheme\\",
  tokens: {
    components: {
      alert: {
        backgroundColor: \\"hsl(210, 5%, 90%)\\",
        padding: \\"0.75rem 1rem\\",
        info: { backgroundColor: \\"hsl(220, 85%, 85%)\\" },
        error: { backgroundColor: \\"hsl(0, 75%, 85%)\\" },
        warning: { backgroundColor: \\"hsl(30, 75%, 85%)\\" },
        success: { backgroundColor: \\"hsl(130, 75%, 85%)\\" },
      },
    },
  },
  overrides: [
    {
      colorMode: \\"dark\\",
      tokens: {
        colors: { black: { value: \\"#fff\\" }, white: { value: \\"#000\\" } },
      },
    },
  ],
}"
`;

exports[`react theme renderer tests theme should render the default theme 1`] = `
"/* eslint-disable */
import { createTheme, defaultTheme } from \\"@aws-amplify/ui-react\\";
export default createTheme(defaultTheme);
"
`;

exports[`react theme renderer tests theme should render the theme 1`] = `
"/* eslint-disable */
import { createTheme } from \\"@aws-amplify/ui-react\\";
export default createTheme({
  name: \\"MyTheme\\",
  tokens: {
    components: {
      alert: {
        backgroundColor: \\"hsl(210, 5%, 90%)\\",
        padding: \\"0.75rem 1rem\\",
        info: { backgroundColor: \\"hsl(220, 85%, 85%)\\" },
        error: { backgroundColor: \\"hsl(0, 75%, 85%)\\" },
        warning: { backgroundColor: \\"hsl(30, 75%, 85%)\\" },
        success: { backgroundColor: \\"hsl(130, 75%, 85%)\\" },
      },
    },
  },
  overrides: [
    {
      colorMode: \\"dark\\",
      tokens: {
        colors: { black: { value: \\"#fff\\" }, white: { value: \\"#000\\" } },
      },
    },
  ],
});
"
`;

exports[`react theme renderer tests theme should render the theme with ES5 1`] = `
"/* eslint-disable */
import { createTheme } from \\"@aws-amplify/ui-react\\";
export default createTheme({
  name: \\"MyTheme\\",
  tokens: {
    components: {
      alert: {
        backgroundColor: \\"hsl(210, 5%, 90%)\\",
        padding: \\"0.75rem 1rem\\",
        info: { backgroundColor: \\"hsl(220, 85%, 85%)\\" },
        error: { backgroundColor: \\"hsl(0, 75%, 85%)\\" },
        warning: { backgroundColor: \\"hsl(30, 75%, 85%)\\" },
        success: { backgroundColor: \\"hsl(130, 75%, 85%)\\" },
      },
    },
  },
  overrides: [
    {
      colorMode: \\"dark\\",
      tokens: {
        colors: { black: { value: \\"#fff\\" }, white: { value: \\"#000\\" } },
      },
    },
  ],
});
"
`;

exports[`react theme renderer tests theme should render the theme with ES5 2`] = `
"declare const _default: any;
export default _default;
"
`;

exports[`react theme renderer tests theme should render the theme with TSX 1`] = `
"/* eslint-disable */
import { createTheme } from \\"@aws-amplify/ui-react\\";
export default createTheme({
  name: \\"MyTheme\\",
  tokens: {
    components: {
      alert: {
        backgroundColor: \\"hsl(210, 5%, 90%)\\",
        padding: \\"0.75rem 1rem\\",
        info: { backgroundColor: \\"hsl(220, 85%, 85%)\\" },
        error: { backgroundColor: \\"hsl(0, 75%, 85%)\\" },
        warning: { backgroundColor: \\"hsl(30, 75%, 85%)\\" },
        success: { backgroundColor: \\"hsl(130, 75%, 85%)\\" },
      },
    },
  },
  overrides: [
    {
      colorMode: \\"dark\\",
      tokens: {
        colors: { black: { value: \\"#fff\\" }, white: { value: \\"#000\\" } },
      },
    },
  ],
});
"
`;
